oauth2/oauth2-client/src/test/java/org/springframework/security/oauth2/client/userinfo/OAuth2UserRequestEntityConverterTests.java

```diff
@@ -0,0 +1,125 @@
+				.clientId("client-1")
+				.clientSecret("secret")
+				.clientAuthenticationMethod(ClientAuthenticationMethod.BASIC)
+				.authorizationGrantType(AuthorizationGrantType.AUTHORIZATION_CODE)
+				.redirectUriTemplate("https://client.com/callback/client-1")
+				.scope("read", "write")
+				.authorizationUri("https://provider.com/oauth2/authorize")
+				.tokenUri("https://provider.com/oauth2/token")
+				.userInfoUri("https://provider.com/user")
+				.userInfoAuthenticationMethod(AuthenticationMethod.HEADER)
+				.userNameAttributeName("id")
+				.build();
+		OAuth2AccessToken accessToken = new OAuth2AccessToken(
+				OAuth2AccessToken.TokenType.BEARER, "access-token-1234", Instant.now(),
+				Instant.now().plusSeconds(3600), new LinkedHashSet<>(Arrays.asList("read", "write")));
+		this.userRequest = new OAuth2UserRequest(clientRegistration, accessToken);
+	}
+
+	@SuppressWarnings("unchecked")
+	@Test
+	public void convertWhenAuthenticationMethodHeaderThenGetRequest() {
+		RequestEntity<?> requestEntity = this.converter.convert(this.userRequest);
+
+		ClientRegistration clientRegistration = this.userRequest.getClientRegistration();
+
+		assertThat(requestEntity.getMethod()).isEqualTo(HttpMethod.GET);
+		assertThat(requestEntity.getUrl().toASCIIString()).isEqualTo(
+				clientRegistration.getProviderDetails().getUserInfoEndpoint().getUri());
+
+		HttpHeaders headers = requestEntity.getHeaders();
+		assertThat(headers.getAccept()).contains(MediaType.APPLICATION_JSON_UTF8);
+		assertThat(headers.getFirst(HttpHeaders.AUTHORIZATION)).isEqualTo(
+				"Bearer " + this.userRequest.getAccessToken().getTokenValue());
+	}
+
+	@SuppressWarnings("unchecked")
+	@Test
+	public void convertWhenAuthenticationMethodFormThenPostRequest() {
+		ClientRegistration clientRegistration = this.from(this.userRequest.getClientRegistration())
+				.userInfoAuthenticationMethod(AuthenticationMethod.FORM)
+				.build();
+		OAuth2UserRequest userRequest = new OAuth2UserRequest(
+				clientRegistration, this.userRequest.getAccessToken());
+
+		RequestEntity<?> requestEntity = this.converter.convert(userRequest);
+
+		assertThat(requestEntity.getMethod()).isEqualTo(HttpMethod.POST);
+		assertThat(requestEntity.getUrl().toASCIIString()).isEqualTo(
+				clientRegistration.getProviderDetails().getUserInfoEndpoint().getUri());
+
+		HttpHeaders headers = requestEntity.getHeaders();
+		assertThat(headers.getAccept()).contains(MediaType.APPLICATION_JSON_UTF8);
+		assertThat(headers.getContentType()).isEqualTo(
+				MediaType.valueOf(APPLICATION_FORM_URLENCODED_VALUE + ";charset=UTF-8"));
+
+		MultiValueMap<String, String> formParameters = (MultiValueMap<String, String>) requestEntity.getBody();
+		assertThat(formParameters.getFirst(OAuth2ParameterNames.ACCESS_TOKEN)).isEqualTo(
+				this.userRequest.getAccessToken().getTokenValue());
+	}
+
+	private ClientRegistration.Builder from(ClientRegistration registration) {
```

Comment:
You could remove this method by creating a ClientRegistration.Builder as a member variable and then you can modify it in the tests before building. Alternatively, you could also directly use TestClientRegistrations and modify it. I'd probably prefer the layer of indirection with the member variable so refactoring would be easier.